pkg/lint/lint.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright Suse LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lint
+
+import (
+	"github.com/rancher-sandbox/hypper/pkg/lint/rules"
+	"helm.sh/helm/v3/pkg/lint/support"
+	"path/filepath"
+)
+
+// All runs all of the available linters on the given base directory.
+func All(basedir string) support.Linter {
+	// Using abs path to get directory context
+	chartDir, _ := filepath.Abs(basedir)
+
+	linter := support.Linter{ChartDir: chartDir}
+	rules.Annotations(&linter)
```

Comment:
Shouldn't we be extending Helm's `pkg/lint`, instead of having our own without the Helm checks?
That way it would be more upstreamable.

I would go with having a `pkg/lint/lint.go.All()` that calls the Helm rules, by importing them. And then `pkg/lint/rules/ ` would keep containing only Hypper rules.
(This means that pkg/action/lint.go doesn't need to call Helm's linters and then Hypper's, of course).

Reply:
Well the `pkg/lint` from helm only has the `All` method, everything is under the `pkg/action/lint` and Im extending that on `pkg/action/lint` already.

Plus changing the `All` method means modifying the signature of that so a full reqork of other pieces.

Not sure how that would be better to what is currently in the PR. Maybe Im just not understanding what you mean?

Reply:
My idea was that `pkg/lint.All()` should call all the Helm rules + the Hypper ones, so:
https://github.com/helm/helm/blob/167aac70832d3a384f65f9745335e9fb40169dc2/pkg/lint/lint.go#L32-L35
+ `rules.Annotations(&linter)`.

That way we don't need to iterate twice for Helm and Hypper rules in `pkg/action/lint.go.Run()`. In fact, we may not need to re-implement `pkg/action/lint.go.Run()` and `pkg/action/lint.go.lintAll()` at all, and then we would be dropping its tests too.

That way we only maintain the Hypper specific tests for `pkg/lint/rules`. And it may very well be that we only need changes there in the future even. Which would make upstreaming `pkg/lint/rules/*` easier.

At the very least, I think we should just call all the Helm rules in `pkg/lint.All()` instead of `pkg/action/lint.go.Run()`.

Reply:
but `pkg/action/lint.go.lintChart` is the one calling the `pkg/lint/lint.go.All()` ?

So we need to override the `pkg/action/lint::Linter.Run` in any case so we can run our rules. Otherwise there is no of hijacking the linter


I still dont see how to make it otherwise. 
The only way is removing the call to the helm linter (which I would not do, reasons below), importing the rules, and calling them in our side just above the hypper ones (pkg/lint/lint.go.All())

My reasons for not wanting to bring the rules are that if something changes in the linter internal code or rules then we may miss it, while the API for the linter is not gonna change ever, we call the linter, it return the results, we dont care what they do internally, if they added, removed or dropped rules, if they now do some mangling to their rules, we only care of running our rules.

Im not against it, its just doesn't seem like a good idea.

Reply:
updated the PR so it now does this.

Reply:
yay! thanks!